python/dask_cudf/dask_cudf/io/orc.py

```diff
@@ -23,16 +25,27 @@ def read_orc(path, **kwargs):
     cudf.read_orc
     """
 
-    filenames = sorted(glob(str(path)))
     name = "read-orc-" + tokenize(path, **kwargs)
-
-    meta = cudf.read_orc(filenames[0], **kwargs)
-
-    graph = {
-        (name, i): (apply, cudf.read_orc, [fn], kwargs)
-        for i, fn in enumerate(filenames)
-    }
-
-    divisions = [None] * (len(filenames) + 1)
-
-    return dd.core.new_dd_object(graph, name, meta, divisions)
+    dsk = {}
+    if "://" in str(path):
+        files = open_files(path)
+        with files[0] as fn:
```

Comment:
The pronoun `fn` is often used for filename, while here it's more used as a file object itself.  It might be worth replacing these with `f`